doc/src/programmers-guide.md

```diff
@@ -1043,13 +1043,16 @@ The `esp:get_mac/1` function can be used to retrieve the network Media Access Co
 
 ## Peripherals
 
-The AtomVM virtual machine and libraries support APIs for interfacing with peripheral devices connected to the ESP32.  This section provides information about these APIs.
+The AtomVM virtual machine and libraries support APIs for interfacing with peripheral devices connected to the ESP32 and other supported microcontrollers.  This section provides information about these APIs.  Unless otherwise stated the documentation for these peripherals is specific to the ESP32, most peripherals are not yet supported on rp2040 or stm32 devices - but work is on-going to expand support on these platforms.
 
 ### GPIO
 
+The GPIO peripheral has nif support on all platforms. One notable difference on the STM32 platform is that `Pin`s are defined as a tuple consisting of the bank (a.k.a. port) and pin number.  For example a pin labeled PB7 on your board would be `{b,7}`.
+
 You can read and write digital values on GPIO pins using the `gpio` module, using the `digital_read/1` and `digital_write/2` functions.  You must first set the direction of the pin using the `gpio:set_direction/2` function, using `input` or `output` as the direction parameter.
 
 To read the value of a GPIO pin (`high` or `low`), use `gpio:digital_read/1`:
+For ESP32 or RP2040:
```

Comment:
The formatting might look strange in the generated docs (two sentences with colons)

Reply:
Yes. You are right, and I realized just a day or two ago that the rp2040 has a slightly different initialization. I will rework this a little. I probably should be leaving the rp2040 documentation to a separate PR, but since it so closely follows the esp32, I was really just trying to differentiate the stm32 from the others.